cylc/flow/workflow_files.py

```diff
@@ -512,17 +512,12 @@ def detect_old_contact_file(reg: str, contact_data=None) -> None:
     else:
         # ... the process isn't running so the contact file is out of date
         # remove it
-        try:
+        with suppress(OSError):
             LOG.info(
                 f'Removing contact file for {reg}'
                 ' (workflow no longer running).'
             )
             os.unlink(fname)
-            return
-        except OSError as exc:
-            raise CylcError(
-                f'Failed to remove old contact file: "{fname}"\n{exc}'
-            )
```

Comment:
Do we not want to keep a warning at least about this? What if there is an `OSError` due to a reason that isn't the file no longer existing?

Reply:
This will most likely be a non-interactive command so there isn't much benefit to raising a warning. We could check if the contact file still exists just before removal and allow the OS error to be raised otherwise?

Reply:
I meant a `LOG.warning()`

Btw, just noticed the docstring will be out of date with this change

Reply:
The contact file is loaded at the start of this function, if it is removed before we try to remove it at the end of the function that is of no consequence so I don't think there's a good reason to log this. Also, this will most likely be a non-interactive command.

Have updated the docstring.

Reply:
> if it is removed before we try to remove it at the end of the function that is of no consequence so I don't think there's a good reason to log this

But I'm talking about an `OSError` that is not an instance of `FileNotFoundError`, e.g. what if some process still has the contact file open?

Just want to check we're happy that's not something to worry about, because if it were to happen, the logging will confusingly say the contact file was removed when it wasn't.

Reply:
Ok, I've change it so that the `removing contact file` message is now `removed contact file` and only shown on success AND logged any non `FileNotFoundError` instance of `OSError` (this log will probably go to a random `job.err` file, but heh, why not).